modules/document_repository/php/files.class.inc

```diff
@@ -62,17 +62,25 @@ class Files extends \NDB_Page
         case "POST":
             return $this->uploadDocFile($request);
         case "PUT":
-            $tmp = $request->getBody();
-            $arr = (array)json_decode($tmp);
-            return $this->editDocFile($arr);
+            return $this->editDocFile($request);
         case "DELETE":
             $id = basename($request->getUri()->getPath());
             $this->deleteFile($id);
             return (new \LORIS\Http\Response\JSON\OK(
                 ['message' => 'File deleted.']
             ));
         case "GET":
-            $filename   = urldecode(basename($request->getUri()->getPath()));
+            $filename = urldecode(basename($request->getUri()->getPath()));
+            if (is_numeric($filename)) {
+                return (new \LORIS\Http\Response())
+                    ->withHeader("Content-Type", "text/plain")
```

Comment:
I don't understand what this is trying to do.. what does the filename being numeric have to do with anything? What if it's not a plain text file?

Reply:
@driusan I hope the URL will get two kinds of filenames.  [ id(number only) and filename ]
ex. "/document_repository/files/4" will return all the information of Upload Fields for edit_document. 
ex. "/document_repository/files/test.pdf" will return the PDF file.

Reply:
Sorry for the delay in responding, but I don't think that's robust. What if someone uploads a file named "4"?

It would make more sense to have 2 different endpoints like `/document_repository/metadata/id` and `/document_repository/files/name` to handle the 2 different types of data being retrieved (though I don't have any opinions on if the first one should be by name or id.)

Reply:
" /document_repository/Files/meta/id" will return meta data for each file.
" /document_repository/Files/filename" will return the file.

Reply:
I added a new endpoint to handle the meta data.  
" /document_repository/Files/meta/id" will return meta data for each file.
" /document_repository/Files/filename" will return the file.